Add full test coverage for update notifications

apyrgio opened the pull request:
This PR adds some extra tests that were missing from the original #466 PR. The new tests cover all the GUI functionality that was not covered in previous tests. Also, it:

1. Improves the logic that is related to getting info from GitHub.
2. Adds a way to differentiate between a user who clicks on "Cancel" and "X".

deeplow commented on dangerzone/gui/updater.py:
```diff
@@ -182,14 +203,30 @@ def get_latest_info(self) -> UpdateReport:
+            changelog = markdown.markdown(info["body"])
+        except KeyError as e:
+            raise ValueError(
+                f"Missing required fields in JSON response from {self.GH_RELEASE_URL}"
```
Or any exception during the parsing of the markdown

apyrgio replied:
Actually no, we catch only `KeyError` here. The reason is that the `str(e)` of a `KeyError` message is just the key name. That's why I wanted to catch it here and add a bit more context.

Markdown rendering errors are actually caught by the catch-all handler and are reported as is.

deeplow replied:
OK. Sounds good.

deeplow reviewed (approved):
Awesome work getting the coverage of updater.py to basically 100%. I have some comments that are so minor that it would be fine to merge it just as is. However you prefer.

apyrgio commented on dangerzone/gui/logic.py:
```diff
@@ -151,9 +151,12 @@ def __init__(
 
         ok_button = QtWidgets.QPushButton(ok_text)
```
I didn't want to cause a big Git diff, but sure.

apyrgio commented on tests/gui/test_main_window.py:
```diff
@@ -1,70 +1,299 @@
+from . import qt_updater as updater
+from .test_updater import assert_report_equal, default_updater_settings
+
+if typing.TYPE_CHECKING:
+    from PySide2 import QtCore
+else:
+    try:
+        from PySide6 import QtCore
+    except ImportError:
+        from PySide2 import QtCore
```
_(huge facepalm)_ Yes, you're totally right! This makes things much simpler!

apyrgio commented on tests/gui/test_updater.py:
```diff
@@ -288,4 +303,145 @@ def test_update_checks_cooldown(updater: UpdaterThread, mocker: MockerFixture) -
+    mocker: MockerFixture,
+) -> None:
+    """Test that the prompt to enable update checks works properly."""
+    qt_updater.dangerzone.settings.set("updater_last_check", 0)
```
I might put it as "it forces Dangerzone to check again for updates".

apyrgio commented:
@deeplow: I've addressed all of your comments, resolved the conversations, and will merge the PR as soon as the CI tests pass. One thing you should have in mind is that I added an extra commit (fdc53ef), to address a flaky test. Basically, pytest-qt did not use our `QApplication` class, so the tests were not 100% accurate.
